File: 2021/day1_1.cpp
#include <iostream>
#include <fstream>
#include <vector>

using namespace std;

static void solve1_1(ifstream &in) {
    vector<int> nums;
    int x;
    while (in >> x)
        nums.push_back(x);
    int cnt = 0;
    for (int i = 1; i < nums.size(); i++) {
        if (nums[i] > nums[i - 1]) cnt++;
    }
    cout << cnt << endl;
}

static void solve1_2(ifstream &in) {
    vector<int> nums;
    int x;
    while (in >> x)
        nums.push_back(x);
    int cnt = 0;
    for (int i = 3; i < nums.size(); i++) {
        if (nums[i] + nums[i - 1] + nums[i - 2] > nums[i - 1] + nums[i - 2] + nums[i - 3]) cnt++;
    }
    cout << cnt << endl;
}

File: 2021/day1_1.py
f = open('2021/input1.txt')

nums = list()

for line in f.readlines():
    nums.append(int(line))


def solve1_1():
    cnt = 0
    for i in range(1, len(nums)):
        if nums[i] > nums[i - 1]:
            cnt += 1
    print(cnt)


def solve1_2():
    cnt = 0
    for i in range(3, len(nums)):
        if nums[i] + nums[i - 1] + nums[i - 2] > nums[i - 1] + nums[i - 2] + nums[i - 3]:
            cnt += 1
    print(cnt)


solve1_1()
solve1_2()
